File: src/tools/wiki_search.py
import wikipedia

def search_wikipedia(query):
    query = query.replace("wikipedia", "")
    query = query.replace("search", "")
    if query == "":
        return "Please provide a search term."
    try:
        results = wikipedia.summary(query)
        return results
    except Exception as e:
        return f"An error occurred: {e}"

print(search_wikipedia("Python"))

File: src/tools/wikiSearch.ts
import wiki from "wikipedia";

interface WikiResult {
  success: boolean;
  data?: {
    title: string;
    summary: string;
    url: string;
  };
  error?: string;
}

export const searchWikipedia = async (query: string): Promise<WikiResult> => {
    try {
        const searchResults = await wiki.search(query);
        if (!searchResults.results.length) {
            return {
                success: false,
                error: `No results found for "${query}"`
            };
        }
        
        const page = await wiki.page(searchResults.results[0].title);
        const summary = await page.summary();

        return {
            success: true,
            data: {
                title: page.title,
                summary: summary.extract,
                url: `https://en.wikipedia.org/wiki/${encodeURIComponent(page.title)}`
            }
        };
    } catch (error) {
        console.error('Wikipedia search error:', error);
        return {
            success: false,
            error: "An error occurred while searching Wikipedia."
        };
    }
};